ipmanager/hetznerFloatingIpConfigurer.go

```diff
@@ -0,0 +1,193 @@
+package ipmanager
+
+import (
+	"context"
+	"errors"
+	"log"
+	"time"
+
+	"github.com/cybertec-postgresql/vip-manager/vipconfig"
+
+	"github.com/hetznercloud/hcloud-go/hcloud"
+)
+
+type HetznerFloatingIpConfigurer struct {
+	*IPConfiguration
+	client       *hcloud.Client
+	server       *hcloud.Server
+	floatingIpId string
+	cachedState  int
+	lastAPICheck time.Time
+	verbose      bool
+}
+
+func newHetznerFloatingIpConfigurer(config *vipconfig.Config, ipConfig *IPConfiguration) (*HetznerFloatingIpConfigurer, error) {
+	client := hcloud.NewClient(hcloud.WithToken(config.HetznerCloudToken))
+	if client == nil {
+		return nil, errors.New("Failed to connect to hetzner cloud")
+	}
+
+	server, _, err := client.Server.Get(context.Background(), config.HetznerCloudServerId)
+	if err != nil {
+		return nil, err
+	}
+
+	c := &HetznerFloatingIpConfigurer{
+		IPConfiguration: ipConfig,
+		client:          client,
+		server:          server,
+		floatingIpId:    config.HetznerCloudIpId,
+		cachedState:     unknown,
+		lastAPICheck:    time.Unix(0, 0),
+		verbose:         config.Verbose,
+	}
+
+	return c, nil
+}
+
+func (c *HetznerFloatingIpConfigurer) queryAddress() bool {
+	if (time.Since(c.lastAPICheck) / time.Second) > 1 {
```

Comment:
Checking once a minute would be a more sensible starting point. Otherwise, we need to make it configurable.
Looking at the old (hetzner robot api) code, we only bothered to recheck that once every hour.
This is really only a courtesy check, following the (successful) registration of the VIP we can expect it to continue pointing our way, unless somebody tampers with it from the outside, in which case it is mainly their responsibility to fix, not ours.